src/libraries/Microsoft.Extensions.Http/src/DependencyInjection/HttpClientBuilderExtensions.cs

```diff
@@ -398,16 +403,22 @@ public static IHttpClientBuilder ConfigureHttpMessageHandlerBuilder(this IHttpCl
         {
             ReserveClient(builder, typeof(TClient), builder.Name, validateSingleType);
 
-            builder.Services.AddTransient<TClient>(s =>
+            builder.Services.AddTransient<TClient>(s => AddTransientHelper<TClient, TImplementation>(s));
+
+            return builder;
+
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:RequiresUnreferencedCode",
```

Comment:
I don't think it should use `RequiresUnreferencedCode` we typically use that only when it involves the RUC attribute. In this case this is typical dataflow warning (on generic parameters).

Reply:
```suggestion
            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:UnrecognizedReflectionPattern",
```

We use `UnrecognizedReflectionPattern` elsewhere:

https://github.com/dotnet/runtime/blob/d4fc640c49f8de360a4a220dbcf48203be6f42c4/src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs#L72

Reply:
Oh good catch, yeah I usually use that too, this was a copy-paste error 😩